### PR TITLE
ADD USB CDC Transport to stm32cube framework 

### PR DESCRIPTION
**Pull request by Darouez-Oussama**

### CODE REVIEW

File: library.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "homepage" : "https://micro.ros.org",
     "repository": {
         "type": "git",
-        "url": "https://github.com/marecustos/micro_ros_platformio#stm32cube"
+        "url": "https://github.com/marecustos/micro_ros_platformio#USB-CDC"
```
Comment (bayrem-gharsellaoui) [inline library.json]: `USB-CDC` -> `stm32cube`

File: platform_code/stm32cube/usb/micro_ros_transport.cpp [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+#include <stdbool.h>
+#include <sys/time.h>
+
+#include <uxr/client/profile/transport/custom/custom_transport.h>
+
+extern "C" {
+#ifdef MICRO_ROS_TRANSPORT_USB_CDC 
+
+//USB Device Handler 
+extern PCD_HandleTypeDef hpcd_USB_OTG_FS;
```
Comment (bayrem-gharsellaoui) [inline platform_code/stm32cube/usb/micro_ros_transport.cpp]: Is this used here?

**Review by bayrem-gharsellaoui — CHANGES_REQUESTED** [review]

**Comment by bayrem-gharsellaoui** [comment]
Can you please base your work on this file and only change what needs to be changed?

https://github.com/micro-ROS/micro_ros_stm32cubemx_utils/blob/jazzy/extra_sources/microros_transports/usb_cdc_transport.c

File: platform_code/stm32cube/usb/micro_ros_transport.cpp (hunk 2 above)
Comment (Darouez-Oussama) [reply platform_code/stm32cube/usb/micro_ros_transport.cpp]: no it is not used i will remove it

**Comment by Darouez-Oussama** [comment]
i have changed #USB-CDC to #stm32cube 
and removed the extern PCD_HandleTypeDef hpcd_USB_OTG_FS; because it is not used

**Comment by bayrem-gharsellaoui** [comment]
> Can you please base your work on this file and only change what needs to be changed?
> 
> https://github.com/micro-ROS/micro_ros_stm32cubemx_utils/blob/jazzy/extra_sources/microros_transports/usb_cdc_transport.c

@Darouez-Oussama

**Comment by Darouez-Oussama** [comment]
I did not understand this . 
do i go to this link and enter github dev make the changes needed and commit ? 
https://github.com/marecustos/micro_ros_platformio/blob/4251724ec5cbc77c7dd20189cc1a6991252ac897/platform_code/stm32cube/usb/micro_ros_transport.cpp

**Review by bayrem-gharsellaoui — APPROVED** [review]

**Pull request merged**